qlib/data/data.py

```diff
@@ -177,46 +166,19 @@ def _get_calendar(self, freq, freq_sam=None, future=False):
         dict
             dict composed by timestamp as key and index as value for fast search.
         """
-        flag = f"{freq}_sam_{freq_sam}_future_{future}"
+        flag = f"{freq}_future_{future}"
         if flag in H["c"]:
             _calendar, _calendar_index = H["c"][flag]
-            return _calendar, _calendar_index
         else:
-            flag_raw = f"{freq}_sam_{None}_future_{future}"
-            if flag_raw in H["c"]:
-                _calendar, _calendar_index = H["c"][flag_raw]
-            else:
-                _calendar = np.array(self.load_calendar(freq, future))
-                _calendar_index = {x: i for i, x in enumerate(_calendar)}  # for fast search
-                H["c"][flag_raw] = _calendar, _calendar_index
-
-            if freq_sam is None:
-                return _calendar, _calendar_index
-            else:
-                _calendar_sam = resam_calendar(_calendar, freq, freq_sam)
-                _calendar_sam_index = {x: i for i, x in enumerate(_calendar_sam)}
-                H["c"][flag] = _calendar_sam, _calendar_sam_index
-                return _calendar_sam, _calendar_sam_index
+            _calendar = np.array(self.load_calendar(freq, future))
+            _calendar_index = {x: i for i, x in enumerate(_calendar)}  # for fast search
+            H["c"][flag] = _calendar, _calendar_index
+        return _calendar, _calendar_index
 
     def _uri(self, start_time, end_time, freq, future=False):
         """Get the uri of calendar generation task."""
         return hash_args(start_time, end_time, freq, future)
 
-    def load_calendar(self, freq, future):
```

Comment:
I think this method should be included in the base class
![image](https://user-images.githubusercontent.com/465606/133203566-8660a6f6-f7de-4d36-a303-0ad77924ecc6.png)